x-pack/test/fleet_api_integration/apis/agents/privileges.ts

```diff
@@ -0,0 +1,186 @@
+import { runPrivilegeTests } from '../../privileges_helpers';
+import { testUsers } from '../test_users';
+
+export default function ({ getService }: FtrProviderContext) {
+  const esArchiver = getService('esArchiver');
+  const supertestWithoutAuth = getService('supertestWithoutAuth');
+  const supertest = getService('supertest');
+  const es = getService('es');
+  let elasticAgentpkgVersion: string;
+
+  describe('fleet_agents_api_privileges', () => {
+    before(async () => {
+      await esArchiver.loadIfNeeded('x-pack/test/functional/es_archives/fleet/agents');
+      const getPkRes = await supertest
+        .get(`/api/fleet/epm/packages/${FLEET_ELASTIC_AGENT_PACKAGE}`)
+        .set('kbn-xsrf', 'xxxx')
+        .expect(200);
+      elasticAgentpkgVersion = getPkRes.body.item.version;
+      // Install latest version of the package
+      await supertest
+        .post(`/api/fleet/epm/packages/${FLEET_ELASTIC_AGENT_PACKAGE}/${elasticAgentpkgVersion}`)
+        .send({
+          force: true,
+        })
+        .set('kbn-xsrf', 'xxxx')
+        .expect(200);
+    });
+    after(async () => {
+      await esArchiver.unload('x-pack/test/functional/es_archives/fleet/agents');
+      await supertest
+        .delete(`/api/fleet/epm/packages/${FLEET_ELASTIC_AGENT_PACKAGE}/${elasticAgentpkgVersion}`)
+        .set('kbn-xsrf', 'xxxx');
+      await es.transport
+        .request({
+          method: 'DELETE',
+          path: `/_data_stream/metrics-elastic_agent.elastic_agent-default`,
+        })
+        .catch(() => {});
+    });
+
+    const READ_SCENARIOS = [
+      {
+        user: testUsers.fleet_all_only,
+        expect: 200,
+      },
+      {
+        user: testUsers.fleet_read_only,
+        expect: 200,
+      },
+      {
+        user: testUsers.fleet_agents_read_only,
+        expect: 200,
+      },
+      {
+        user: testUsers.fleet_no_access,
+        expect: 403,
+      },
+      {
+        user: testUsers.fleet_minimal_all_only,
+        expect: 403,
+      },
+      {
+        user: testUsers.fleet_minimal_read_only,
+        expect: 403,
+      },
+      {
+        user: testUsers.fleet_settings_read_only,
+        expect: 403,
+      },
+    ];
+
+    const ALL_SCENARIOS = [
+      {
+        user: testUsers.fleet_all_only,
+        expect: 200,
+      },
+      {
+        user: testUsers.fleet_agents_all_only,
+        expect: 200,
+      },
+      {
+        user: testUsers.fleet_agents_read_only,
+        expect: 403,
+      },
+      {
+        user: testUsers.fleet_read_only,
+        expect: 403,
+      },
+      {
+        user: testUsers.fleet_no_access,
+        expect: 403,
+      },
+      {
+        user: testUsers.fleet_minimal_all_only,
+        expect: 403,
+      },
+      {
+        user: testUsers.fleet_minimal_read_only,
+        expect: 403,
+      },
+      {
+        user: testUsers.fleet_settings_read_only,
+        expect: 403,
+      },
+    ];
+
+    let agentDoc: any;
+
+    const ROUTES = [
```

Comment:
@kpollich it is what you add in mind for the API integration tests? this generate tests like this for each route
<img width="993" alt="Screenshot 2024-03-27 at 1 48 05 PM" src="https://github.com/elastic/kibana/assets/1336873/3339c4f9-b6ef-48a2-ab42-2cd741207f9e">

Reply:
Yes exactly this looks great!